client/src/main/java/com/linecorp/decaton/client/DecatonClient.java

```diff
@@ -46,6 +47,25 @@
      */
     CompletableFuture<PutTaskResult> put(String key, T task, long timestamp);
 
+    /**
+     * Put a task onto associated decaton queue with specifying arbitrary delay duration.`
+     * @param key the criteria to shuffle and order tasks. null can be specified if it doesn't matters.
+     * @param task an instance of task. Should never be null.
+     * @param delayDuration duration which is used to set delay of task metadata.
+     * @return a {@link CompletableFuture} which represents the result of task put.
+     */
+    CompletableFuture<PutTaskResult> put(String key, T task, Duration delayDuration);
+
+    /**
+     * Put a task onto associated decaton queue with specifying arbitrary timestamp and arbitrary delay.
+     * @param key the criteria to shuffle and order tasks. null can be specified if it doesn't matters.
+     * @param task an instance of task. Should never be null.
+     * @param timestamp milliseconds precision timestamp which is to be used to set timestamp of task metadata.
+     * @param delayInMillis milliseconds which is used to set delay duration of task metadata.
+     * @return a {@link CompletableFuture} which represents the result of task put.
+     */
+    CompletableFuture<PutTaskResult> put(String key, T task, long timestamp, long delayInMillis);
```

Comment:
Thanks for the PR! 
I know that there are several people who're building a task metadata manually and produce to enable scheduled execution (which we added just for retry but unexpectedly leveraged for other purpose) so I think this change will be helpful for many users 👍 

Just one concern is, if we keep adding more overloads of `put` methods along with the list of metadata properties to support that'll make the interface to have massive amount of overloads that combines many different sets of metadata properties to customize.
So at this point I think we should introduce a interface like:
```java
     @Value
     @Builder
     class TaskMetadata {
         Long timestamp;
         Long executeTime;
     }
     CompletableFuture<PutTaskResult> put(String key, T task, TaskMetadata overrideMetadata); 
```

and let users to use it in the way:
```java
decatonClient.put(..., TaskMetadata.builder().setScheduledTime(1234).build());
```

WDYT?

Reply:
Yes, that's better.  I was planning to do that when adding the new methods,  but since there are already some put methods defined with the timeStamp, finally I decided to add methods only with scheduledTime.  In view of the possible usage and possible modification on taskMetaData in the future, your recommanded way is much better (although that kind of definition is also able to be used to provide same function as the existing methods with timestamp as parameter).